client/my-sites/controller.js

```diff
@@ -297,7 +299,7 @@ export function siteSelection( context, next ) {
 	const { getState, dispatch } = getStore( context );
 	const siteFragment = context.params.site || getSiteFragment( context.path );
 	const basePath = sectionify( context.path, siteFragment );
-	const currentUser = user.get();
+	const currentUser = getCurrentUser( getState() );
```

Comment:
When the user is logged out, `user.get()` returns `false`, while the `getCurrentUser` selector returns `null`. That's much bigger difference than it seems, because booleans can have properties 😄 Therefore `currentUser.visible_site_count` evaluates to `undefined` instead of crashing. `null` value is not so lucky.

Changing the `hasOneSite` check to `currentSite && currentSite.visible_site_count === 1` should fix all the problems and make this patch mergeable again.